### ISSUE TITLE
Assembly of small variants misinterprets events.

### ISSUE THREAD

**Issue by d-cameron** [PapenfussLab/gridss#320]
Assembly of small events misinterprets events due to flanking homology and lack of DUP handling in SW alignment. In particular:

- DUPs are reported as INS
- DELs in regions where the ref is a tandem duplication are reported as SNVs, not DELs

Solution:
- Convert INS CIGAR elements to DUP when the inserted sequence matches the ref.
  - what's the min size for this - 8bp?)
- Do both new (whole-read) and old (only breakend sequence) realignment and work out which one is best
- Maybe switch to using minimap2?

**Comment by DarioS** [comment]
Does it mean users should temporarily discard variants less than 50 bases wide to avoid this problem?

**Comment by d-cameron** [comment]
If your assembly is long enough then it'll happen for events greater than 50bp. This is a general issue that's not restricted to GRIDSS. This shows up particularly often when comparing long and short read calls in the 50-300bp range as the long read caller will also always report an insertion, and the short read caller a duplication.

This enhancement tag is because GRIDSS is sometimes acting more like a long read caller, and the bug tag is because the GRIDSS supporting evidence sometimes gets split between the ins and the dup thus the variant call is filtered.

This came up when I was benchmarking GRIDSS2 on HG002 and noticed GRIDSS1 was outperforming GRIDSS2 for certain event types. During GRIDSS2, I switched from aligning only the assembly breakend bases to realigning the entire assembly contig to eliminate a class of false positives, I just didn't notice this class of false negatives till now.

TLDR: no need to filter, it's just GRIDSS2 is a less sensitive for small DUPs than GRIDSS1.
